### DISCUSSION TITLE
Simpler nginx install

### DISCUSSION THREAD

**Discussion by mariusa** [piku/piku#288] [Announcements]
Hi,

Nowadays `nginx` already comes with `/etc/nginx/nginx.conf` which has a default server running on port 80.
This is overwritten by https://raw.githubusercontent.com/piku/piku/master/nginx.default.dist

Instead, the only line that needs to be added is
`include /home/piku/.piku/nginx/*.conf;`

Since `nginx.conf` also has
`include /usr/share/nginx/modules/*.conf;`

I think it's best to leave that alone and simply create a `/usr/share/nginx/modules/piku.conf` file which has
`include /home/piku/.piku/nginx/*.conf;`

**Comment by chr15m** [comment]
Hi @mariusa. Cool idea. Can you qualify "nowadays"? I think it's important to make sure older boxes are supported to some degree.

**Comment by mariusa** [comment]
I don't have an Ubuntu OS around, but I see this https://github.com/piku/piku/blob/master/docs/ubuntu-18.04-bionic.md
was already updated to use the simpler distro version of uwsgi 

I wouldn't like to get stuck to old ways and not being able to use it (easily) on a new system. Similar to software libs, a branch & git release could be created with the current version, then start evolving the main branch to newer systems.
This way one can choose what is best for each case.

**Comment by chr15m** [comment]
Ah yep of course. I'm going to set up tests on `piku-bootstrap` which will help with stuff like this.

**Comment by rcarmo** [comment]
I can test that as well. TBH I see Piku boxes as single-purpose things and don't mind overwriting `nginx.conf` (especially since things get really hairy when you start adding multiple domains and caching), but if it works I'm fine with it.

**Comment by rcarmo** [comment]
Moving this to discussions, we can continue there.

**Comment by mariusa** [comment]
Done in INSTALL-centos (PR open)

There's an issue for me though. After doing `git push piku main` I get
```
remote: Error: [nginx config] b'2023/03/23 15:25:11 [emerg] 21913#21913: "upstream" directive is not allowed here in /home/piku/.piku/nginx/app.conf:3\n'
remote: Warning: removing broken nginx config.
```

Found a solution here https://stackoverflow.com/questions/53454739/nginx-upstream-failure-configuration-file
but I can't test it as the app nginx.conf is removed from .piku/nginx/
so I don't know what to edit to fix this. How could I tell piku to generate the nginx config file for an app?

**Comment by mariusa** [comment]
solved in latest PR, it was my setup
